src/run.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as logging from 'plylog';
+import * as resolve from 'resolve';
+import * as updateNotifier from 'update-notifier';
+import * as cliTypeOnly from './polymer-cli';
+
+const packageJson = require('../package.json');
+
+const logger = logging.getLogger('cli.main');
+
+
+// Update Notifier: Asynchronously check for package updates and, if needed,
+// notify on the next time the CLI is run.
+// See https://github.com/yeoman/update-notifier#how for how this works.
+updateNotifier({pkg: packageJson}).notify();
+
+resolve('polymer-cli', {basedir: process.cwd()}, async(_error, path) => {
```

Comment:
We _should_ consider just removing this feature, but only very carefully.

It was a bit cargo-culted from WCT and polybuild, where the idea is that they're both a CLI and a library, and a global install of the CLI should defer to the local version of the library if present. Gulp essentially does this, except that it requires a local install - something we did and reverted for WCT 6.

I don't think the CLI is installed much as a library... though it could be. The key factor for keeping this feature is - should we let a project pin the version of the CLI it uses by installing it as a dev dependency. Given the rapid changes we've gone through lately, that seems like a prudent feature to me. We could be a little more clear about the working here though: print the version of CLI being used at each invocation, do update checks on both global and local...

Either way, we shouldn't change this in this PR.